### PR TITLE
fix: handle more 'workspace:*' specifiers in package.json

### PR DESCRIPTION
**Pull request by bartlomieju**
I will open PRs to all relevant crates to bump `deno_package_json` before merging this one

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -344,8 +360,15 @@ impl PackageJson {
       value: &str,
     ) -> Result<PackageJsonDepValue, PackageJsonDepValueParseError> {
       if let Some(workspace_key) = value.strip_prefix("workspace:") {
-        let version_req = VersionReq::parse_from_npm(workspace_key)?;
-        return Ok(PackageJsonDepValue::Workspace(version_req));
+        let workspace_req = match workspace_key {
+          "*" => PackageJsonDepWorkspaceReq::Star,
```
Comment (dsherret) [inline src/lib.rs:364]: I think `*` is ok to just represent as a VersionReq

File: src/lib.rs [hunk 2]
```diff
@@ -36,10 +36,22 @@ pub enum PackageJsonDepValueParseError {
   Unsupported { scheme: String },
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum PackageJsonDepWorkspaceReq {
```
Comment (bartlomieju) [inline src/lib.rs:40]: Don't love the name, any suggestions?

Comment (dsherret) [reply src/lib.rs:40]: Seems ok.

**Review by dsherret — APPROVED** [review]
LGTM

**Pull request merged**